### ISSUE TITLE
ProbaV Harvester: update_all configuration option

### ISSUE THREAD

**Issue by JGulic** [NextGeoss/ckanext-nextgeossharvest#180]
The ProbaV harvester doesn't provide an update_all option in the configuration of the harvester.

We need to add this option, in case we need to update something about this harvester, like adding new metadata.

**Comment by georgiana-b** [comment]
@davidcordeiro-deimos I tested the ProbaV harvester as it was implemented in https://github.com/NextGeoss/ckanext-nextgeossharvest/pull/190 and I have a question about how this works.

Right now, after https://github.com/NextGeoss/ckanext-nextgeossharvest/pull/190, if we start a harvester with the following configuration:

```json
{
"start_date":"2018-08-01",
"collection":"PROBAV_S1-TOC_1KM_V001",
"username":"nextgeoss",
"password":"nextgeoss",
"make_private":false,
"update_all":true
}
```

The harvester will behave in the following way:
- on the first run it harvests datasets for the first 2 months 
- from the second run further, it updates the datasets from the previous month and collects new ones from the next month

<img width="1680" alt="Screenshot 2020-02-10 at 17 22 03" src="https://user-images.githubusercontent.com/8862002/74229910-9deb5200-4cc3-11ea-844d-3e39f6fb18c1.png">

I'm just trying to understand if this behaviour is necessary since it doesn't seem to help with our use case.

This feature was needed in order to update historical datasets. So we want to make sure that, if we start a harvester with the same configuration as the one above, it will run normally, one month at a time, updating the datasets if they already exist. 

So the harvester should only move forward, one month at a time, and either create or update the datasets in that month. Can you explain why it is necessary to go over the previous month?

**Comment by joaandrade** [comment]
@georgiana-b this Proba-V harvester is not going over the previous month. The S1 collections have, on the original data source, a product per day. However this product contains a huge number of tiles. On our catalogue, we are considering that each tile is a separated dataset (to avoid a dataset with 330 resources). Also this is the approach preferred by the provider:
**S1 collection -> 1 product per day -> 330 tiles -> 330 datasets per day**

The logic is the same to all the S(x) collections. The difference is the frequancy:
S1 -> daily
S5 -> each 5 days
S10 -> each 10 days
Then you have the L2 collections with multiple products per day (as a regular harvester).

So, what is happening is: the harvester is updating the datasets from the previous day due to the flag update_all=true.If you set it as false, it will only add the new ones. However, since the restart date is the end of the previous day, it will check that day again. This is a mechanism to ensure we catalogue all the data.

Hope this helps.

**Comment by georgiana-b** [comment]
@joaandrade Thank you Joao. That helps a lot. So in that case my question is, why does it need to go over the previous day? 

I'm asking because, as far as I can see, it should always move forward, either creating or updating the dataset it encounters.

**Comment by joaandrade** [comment]
Imagine we harvest today (11/02), the restart date will be 11/02 until tomorrow's datasets are harvested. Tomorrow, before harvesting the new datasets:
restart_date=11/02 so start_date=11/02, end_date=12/02
The harvester will do a query from 11/02 to 12/02. This ensures that no product was missed.

**Comment by georgiana-b** [comment]
I understand that it ensures that no product was missed but it also means the harvester is wasting time reharvesting every day twice.  I understand this is just how they return the results at the source - for every day inclusively - but I would say it makes sense to change the query so that restart_date=end_date to make sure we always get only the last day. Just my opinion.

@JGulic  This explains why ProbaV keeps reharvesting datasets in every run. Let us know what you think about the above.

**Comment by joaandrade** [comment]
The suggested behavior is ok for the S(x) collections. However, for the L2 collections is not secure to do that.

**Comment by JGulic** [comment]
I've merged PR #190 
If we agree I would suggest to close this ticket, and open a new one for further improvements.

**Comment by joaandrade** [comment]
@JGulic, I agree with the suggested approach
